### PR TITLE
[Benchmark] full training benchmarks

### PR DESCRIPTION
**Pull request by albertbou92**
## Description

Describe your changes in detail.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/pytorch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [ ] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: benchmarks/full_trainings/dqn_mnih15_atari.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+        assert noops > 0
+        obs = None
+        for _ in range(noops):
+            obs, _, done, *other = self.env.step(self.noop_action)
+            if done:
+                obs = self.env.reset(**kwargs)
+        return obs
+
+
+class EpisodicLifeEnv(gym.Wrapper):
```
Comment (vmoens) [inline benchmarks/full_trainings/dqn_mnih15_atari.py:74]: Should be a transform

File: benchmarks/full_trainings/dqn_mnih15_atari.py [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+    loss_module = DQNLoss(
+        value_network=model,
+        gamma=gamma,
+        loss_function="l2",
+        delay_value=True,
+        # action_space="categorical"
+    )
+    loss_module.make_value_estimator(gamma=gamma)
+
+    target_net_updater = HardUpdate(
```
Comment (vmoens) [inline benchmarks/full_trainings/dqn_mnih15_atari.py:252]: Use soft one for rainbow

**Comment by vmoens** [comment]
As discussed, I would rather change the examples than put this in benchmark for the following reasons:
1. Benchmarks are to measure speed of low-level components of the lib. Maybe in the future we'll also test full training pipelines but this seems to be oriented toward performance benchmarks in terms of return (not speed)
2. We already have examples and it will be confusing for the users to have 2 examples of a single algo within the lib. 

At the end of the day, our examples have the same scope as these benchmarks: reproducing SOTA results. We can rename examples into "implementations" or anything else if that makes it clearer.

Also, IMO for DQN we should have 1 script for Rainbow (that we need) and 1 config for Rainbow + 1 config for DQN (original paper -- ablation study).

Images should not be hosted in torchrl but on our S3 bucket

**Comment by vmoens** [comment]
Another point: I think each script should be an independent PR :) 
We should also clearly define what metrics we must log (in order for us to make sure that all algos log the same stuff).
Here's a few to get started:
- [ ] Returns
- [ ] Gradient norm
- [ ] Loss values (if they make sense?)
- [ ] Episode length (if it makes sense)

These are just ideas obv

Each script should be accompanied by a `benchmark_algoXX.yaml` that we run on the RC branch before the release to make sure that the algo runs fine. This yaml file is a SOTA implementation: it must be close to the paper results / public results on another lib. Runtime isn't really an issue for this file (we can have another one that is called `fast_blabla`).

@BY571 for context

File: benchmarks/full_trainings/dqn_mnih15_atari.py [hunk 3]
```diff
@@ -0,0 +1,369 @@
...
+
+
+def make_parallel_env(env_name, device, is_test=False):
+    env = ParallelEnv(1, EnvCreator(lambda: make_base_env(env_name, device)))
+    env = TransformedEnv(env)
+    env.append_transform(ToTensorImage())
+    env.append_transform(GrayScale())
+    env.append_transform(Resize(84, 84))
+    env.append_transform(CatFrames(N=4, dim=-3))
+    env.append_transform(RewardSum())
```
Comment (matteobettini) [inline benchmarks/full_trainings/dqn_mnih15_atari.py:115]: reward sum was broken for parallel envs, could have this been the problem?
